### ISSUE TITLE
Adding Layout to logback-ecs-encoder

### ISSUE THREAD

**Issue by dgempiuc** [elastic/ecs-logging-java#144]
If there is an error in incoming requests, I am logging this request and some requests may contain sensitive information, such as  password. 

There is LayoutWrappingEncoder encoder on logback and I am able to add Layout that.

```
        <encoder class="ch.qos.logback.core.encoder.LayoutWrappingEncoder">
            <layout class="org.custom.ecs.layout.MaskingPatternLayout">
                <maskPattern>\"oldPassword\"\s*:\s*\"(.*?)\"</maskPattern> <!-- password pattern -->
                <maskPattern>\"newPassword\"\s*:\s*\"(.*?)\"</maskPattern> <!-- password pattern -->
                <pattern>${CONSOLE_LOG_PATTERN}</pattern>
            </layout>
        </encoder>
```

payload={
  "newPassword": "\********\",
  "oldPassword": "********",
  "smsCode": "123456"
}

But, I am using EcsEncoder in production environment, but there is no option to add Layout in this encoder. So, I can't mask fields. Is there anyway to do that or is there any plan to add this?

**Comment by dgempiuc** [comment]
created pr https://github.com/elastic/ecs-logging-java/pull/145

**Comment by Tokimimo** [comment]
Any chance such a function will be added? Or am I maybe missing something to achieve the above with the current encoder or available resources?

**Comment by michalgutkowski** [comment]
@Tokimimo  IMO it works as expected after recent changes in #145: 

     _<appender name="ECS_JSON_CONSOLE" class="ch.qos.logback.core.ConsoleAppender">
        <encoder class="co.elastic.logging.logback.EcsEncoder">
            <messageLayout class="ch.qos.logback.classic.PatternLayout">
                <pattern>[%d{HH:mm:ss.SSS}] %t %-5p %c  - %m%n %rEx{full, org.springframework}</pattern>
            </messageLayout>
        </encoder>
    </appender>_

**Comment by rdifrango** [comment]
This was closed by #220 which was built upon #145 and has since been merged.  @felixbarny and @mgreau we should close this issue now that it's been resolved.
